pkg/gardenadm/cmd/token/utils/compute.go

```diff
@@ -43,6 +45,41 @@ func CreateBootstrapToken(ctx context.Context, clientSet kubernetes.Interface, o
 		return fmt.Errorf("failed creating a bootstrap token secret: %w", err)
 	}
 
-	fmt.Fprintln(opts.IOStreams.Out, bootstraptoken.FromSecretData(secret.Data))
+	if !opts.PrintJoinCommand {
+		fmt.Fprintln(opts.Out, bootstraptoken.FromSecretData(secret.Data))
+	} else {
+		output, err := printJoinCommand(ctx, opts, clientSet, secret)
+		if err != nil {
+			return fmt.Errorf("failed computing join command: %w", err)
+		}
+		fmt.Fprint(opts.Out, output)
+	}
+
 	return nil
 }
+
+func printJoinCommand(ctx context.Context, opts *Options, clientSet kubernetes.Interface, bootstrapTokenSecret *corev1.Secret) (string, error) {
+	secretList := &corev1.SecretList{}
+	if err := clientSet.Client().List(ctx, secretList, client.InNamespace(metav1.NamespaceSystem), client.MatchingLabels{
+		v1beta1constants.GardenRole:      v1beta1constants.GardenRoleOperatingSystemConfig,
+		v1beta1constants.LabelWorkerPool: opts.WorkerPoolName,
+	}); err != nil {
+		return "", fmt.Errorf("failed listing gardener-node-agent secrets: %w", err)
+	}
+
+	if len(secretList.Items) == 0 {
+		return "", fmt.Errorf("no gardener-node-agent secrets found for worker pool %q", opts.WorkerPoolName)
+	}
+
+	gardenerNodeAgentSecret := secretList.Items[0]
+	if len(secretList.Items) > 1 {
+		opts.Log.V(1).Info("Multiple gardener-node-agent secrets found, using the first one", "secretName", gardenerNodeAgentSecret.Name)
+	}
+
+	return fmt.Sprintf(`gardenadm join --bootstrap-token %s --ca-certificate "%s" --gardener-node-agent-secret-name %s %s`,
+		bootstraptoken.FromSecretData(bootstrapTokenSecret.Data),
+		utils.EncodeBase64(clientSet.RESTConfig().CAData),
+		gardenerNodeAgentSecret.Name,
+		clientSet.RESTConfig().Host,
```

Comment:
This returns `localhost` 👀

Reply:
Yes, with https://github.com/gardener/gardener/pull/11942/commits/e83da1cf2fc9a8791cdd6a1c868a627e174d1a72, it will return a useable name. :)